### PR TITLE
Issue #20368: Preload EJB pools for CRIU

### PR DESCRIPTION
**Pull request by tkburroughs**
Default EJB stateless beans to preload the EJB pool during application start
for server checkpoint. Preload will not occur if StartAtAppStart disabled.

for #20368

### CODE REVIEW

**Comment by tkburroughs** [comment]
#build

**Comment by tkburroughs** [comment]
#build

File: dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ejs/csi/EJBApplicationMetaData.java [hunk 1]
```diff
@@ -198,25 +202,38 @@ public class EJBApplicationMetaData {
...
+    /**
+     * List of Runnable instances that will preload bean pools for those stateless beans that
+     * have been configured with a hard minimum pool size. Additionally, stateless bean pools
+     * may be preloaded when the server has been started for taking a checkpoint.
+     *
+     * Bean pools will be preloaded at the end of application start after all startup bean
+     * post construct methods have been called and the application will be unblocked for work
+     * to to ensure all beans in the application have been identified (in case there are bean
+     * references) and to ensure the application state has been properly initialized before
+     * running any additional application code that may occur during the preload.
```
Comment (brideck) [inline dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ejs/csi/EJBApplicationMetaData.java]: This comment is repeated at line 435 verbatim, and it took me some effort to parse it.  Perhaps restructuring and/or adding punctuation would make this easier to decipher?

Maybe something like:
To ensure that all beans in the application have been identified (in case there are bean references) and that the application's state has been properly initialized (before running any additional application code that may occur during the preload), bean pools will be preloaded at the end of application start, after all startup bean post construct methods have been called, and the application will be unblocked for work.

**Review by brideck — APPROVED** [review]
Functional changes look good, just one comment on readability.

File: dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ejs/csi/EJBApplicationMetaData.java (hunk 1 above)
Comment (tkburroughs) [reply dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ejs/csi/EJBApplicationMetaData.java]: @brideck Agree this is hard to follow.  I've removed the duplicate copy of the comment on the instance variable declaration, and updated the version on the add method similar to your suggestions, as follows:

```
     * To ensure all beans in the application have been identified (in case there are bean
     * references that need to be resolved prior to instance creation) and the application's
     * state has been properly initialized (before running any additional application code that
     * may occur during the preload), bean pools will be preloaded at the end of application
     * start, after all beans have been identified, all startup bean post construct methods
     * have been called, and the application has been unblocked for work.
```

**Comment by tkburroughs** [comment]
#libby build

**Comment by tkburroughs** [comment]
#build

**Review by olendvcook — APPROVED** [review]
